spec.bs

```diff
@@ -248,6 +253,92 @@ If using a centralized mechanism for fetching keys, user agents should have a re
 
 When using a registration process, it is recommended that user agents apply an expiration date to registration requests, to allow for the removal of deprecated or no longer active issuers.
 
+VOPRF Methods {#voprf-methods}
+====================================
+
+This document encodes protocol messages in the TLS presentation language from section 3 of [[!RFC8446]].
+
+When the server is performing the [=BlindEvaluate=] function of the [[!VOPRF]] protocol, the input <dfn>IssueRequest</dfn> and output <dfn>IssueResponse</dfn> are serialized as:
+
+```
+// Scalars are elliptic curve scalars of length Ns (determined by the curve, 48 for P-384).
+// ECPoints are elliptic curve points encoded using the X9.62 Uncompressed point representation (determined by the curve, 97 for P-384).
+
+struct {
+  uint16 count;
+  ECPoint nonces[count];
+} IssueRequest;
+
+struct {
+  ECPoint blinded;
+  ECPoint evaluated;
+} SignedNonce;
+
+struct {
+  Scalar c;
+  Scalar s;
+} DLEQProof;
+
+struct {
+  uint16 issued;
+  uint32 key_id;
+  SignedNonce signed[issued];
+  opaque proof<1..2^16-1>; // Bytestring containing a serialized DLEQProof struct.
+} IssueResponse;
+```
+
+When the server is performing the [=Evaluate=] function of the [[!VOPRF]] protocol, the input <dfn>RedeemRequest</dfn> and output <dfn>RedeemResponse</dfn> are serialized as:
+
+```
+struct {
+  uint32 key_id;
+  opaque nonce<nonce_size>;
+  ECPoint W;
+} Token;
+
+struct {
+  opaque token<1..2^16-1>; // Bytestring containing a serialized Token struct.
```

Comment:
I assume serialization details are explained in RFC8446?

Reply:
The RFC8446 encoding specifies what each of the constructs is encoded as (network byte order, etc). Not sure we need to call out each individual struct member?